ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml

```diff
@@ -16,13 +16,17 @@ chain:
     - name: HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT
       default: "false"
       documentation: "Whether to use the generic CI account or the HyperShift OSD account for the guest clusters infra. For the infra created for the clusters. E.g. For cluster-bot we use the generic CI account"
+    - name: HYPERSHIFT_BASE_DOMAIN
+      default: "ci.hypershift.devcluster.openshift.com"
+      documentation: "The cluster's FQDN will be a subdomain of the base domain."
     commands: |-
       AWS_GUEST_INFRA_CREDENTIALS_FILE="/etc/hypershift-ci-jobs-awscreds/credentials"
-      BASE_DOMAIN=ci.hypershift.devcluster.openshift.com
       
       if [[ $HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT == "true" ]]; then
         AWS_GUEST_INFRA_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
-        BASE_DOMAIN=origin-ci-int-aws.dev.rhcloud.com
+        if [[ $HYPERSHIFT_BASE_DOMAIN != "qe.devcluster.openshift.com" ]]; then
```

Comment:
Can you please don't default the env variable, keep the logic as before this PR, and then just let the environment variable be authoritative if it was set?

The likes of
```
      AWS_GUEST_INFRA_CREDENTIALS_FILE="/etc/hypershift-ci-jobs-awscreds/credentials"
      DEFAULT_BASE_DOMAIN=ci.hypershift.devcluster.openshift.com
      
      if [[ $HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT == "true" ]]; then
        AWS_GUEST_INFRA_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
        DEFAULT_BASE_DOMAIN=origin-ci-int-aws.dev.rhcloud.com
      fi
      
      DOMAIN :=${HYPERSHIFT_DOMAIN:-DEFAULT_BASE_DOMAIN}
```

That way the behaviour is generic

Reply:
okay, thanks for your comment, good idea

Reply:
Ah, it assumes `HYPERSHIFT_BASE_DOMAIN` has the highest weight.  If it is set in jobs, `HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT` won't work any more.

Reply:
Then for existing chains, we need remove all existing env `HYPERSHIFT_BASE_DOMAIN` with default value ? so that it won't affect `HYPERSHIFT_GUEST_INFRA_OCP_ACCOUNT`